src/Variant.cpp

```diff
@@ -25,937 +25,955 @@
+          }
+        }
 
-      left.cast (type_duration);
-      return left._duration <= right._duration;
-    }
-    break;
+        case type_date:
+          if (left.trivial() || right.trivial()) return false;
 
-  case type_date:
-    switch (right._type)
-    {
-    case type_boolean:
-    case type_integer:
-    case type_real:
-    case type_string:
-    case type_date:
-    case type_duration:
-      if (left.trivial () || right.trivial ())
-        return false;
+          left.cast(type_date);
+          return left._date <= right._date;
 
-      right.cast (type_date);
-      return left._date <= right._date;
-    }
-    break;
+        case type_duration:
+          if (left.trivial() || right.trivial()) return false;
+
+          left.cast(type_duration);
+          return left._duration <= right._duration;
+      }
+      break;
 
-  case type_duration:
-    switch (right._type)
-    {
-    case type_boolean:
-    case type_integer:
-    case type_real:
-    case type_string:
     case type_date:
-    case type_duration:
-      if (left.trivial () || right.trivial ())
-        return false;
+      switch (right._type) {
+        case type_boolean:
+        case type_integer:
+        case type_real:
+        case type_string:
+        case type_date:
+        case type_duration:
+          if (left.trivial() || right.trivial()) return false;
+
+          right.cast(type_date);
+          return left._date <= right._date;
+      }
+      break;
 
-      right.cast (type_duration);
-      return left._duration <= right._duration;
-    }
-    break;
+    case type_duration:
+      switch (right._type) {
+        case type_boolean:
+        case type_integer:
+        case type_real:
+        case type_string:
+        case type_date:
+        case type_duration:
+          if (left.trivial() || right.trivial()) return false;
+
+          right.cast(type_duration);
+          return left._duration <= right._duration;
+      }
+      break;
   }
 
   return false;
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-bool Variant::operator> (const Variant& other) const
-{
-  Variant left (*this);
-  Variant right (other);
-
-  if (left._type == type_string)
-    Lexer::dequote (left._string);
-
-  if (right._type == type_string)
-    Lexer::dequote (right._string);
-
-  switch (left._type)
-  {
-  case type_boolean:
-    switch (right._type)
-    {
-    case type_boolean:                             return !left._bool && right._bool;
-    case type_integer:  left.cast (type_integer);  return left._integer  > right._integer;
-    case type_real:     left.cast (type_real);     return left._real     > right._real;
-    case type_string:   left.cast (type_string);   return left._string   > right._string;
-    case type_date:     left.cast (type_date);     return left._date     > right._date;
-    case type_duration: left.cast (type_duration); return left._duration > right._duration;
```

Comment:
There are a number of snippets like this, where non-standard formatting was used in a way that arguably made the code easier to understand. Should we try to retain that? Is that difficult? Worth the effort?

I don't have a particular opinion -- I just want to be sure that this was considered.

Reply:
https://clang.llvm.org/docs/ClangFormatStyleOptions.html#disabling-formatting-on-a-piece-of-code

Yes, definitely it is possible to turn it off for certain regions.

Enclosing the lines with a comment 
```
// clang-format off 
code without formatting

// clang-format on
```

I also don't have any particular opinion.

Reply:
@tbabej I'm curious what you think, as perhaps you wrote some of those nicely-formatted sections. Maybe there are a few that are particularly helpful that we should preserve?

But without any strong opinions I think we should just let clang-format do its thing.